src/DataSets/Forms/Save.component.js

```diff
@@ -8,22 +8,22 @@ import CardText from 'material-ui/Card/CardText';
 const Save = React.createClass({
     mixins: [Translate],
 
-    states: {DATAENTRY: "DATAENTRY", SAVED: "SAVED", SAVE_ERROR: "SAVE_ERROR"},
-
     propTypes: {
         config: React.PropTypes.object,
-        data: React.PropTypes.object,
+        store: React.PropTypes.object,
         state: React.PropTypes.string,
         errors: React.PropTypes.arrayOf(React.PropTypes.string)
     },
 
     render() {
-        const {dataset} = this.props.data;
+        const {dataset} = this.props.store;
         const {state, errors} = this.props;
 
-        if (state == this.states.DATAENTRY) {
+        if (state === "DATAENTRY" || state === "SAVING") {
```

Comment:
@tokland I saw you created a state dictionary in FormSteps component but it is not used here... it would be better if we use a unique dictionary for the states. I guess to pass it as a parameter can be a bit messy but maybe as mixin? What do you think?